sound/soc/sof/intel/hda-dsp.c

```diff
@@ -819,7 +833,6 @@ int hda_dsp_runtime_suspend(struct snd_sof_dev *sdev)
 int hda_dsp_suspend(struct snd_sof_dev *sdev, u32 target_state)
 {
 	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
-	struct hdac_bus *bus = sof_to_bus(sdev);
 	struct pci_dev *pci = to_pci_dev(sdev->dev);
 	const struct sof_dsp_power_state target_dsp_state = {
 		.state = target_state,
```

Comment:
This threw me a bit off in the review. To ensure the behaviour of hda_dsp_suspend() is not changed by the patch, you have to go through many functions (hda_dsp_suspend() -> snd_sof_dsp_set_power_state() -> hda_dsp_set_D0_state()) and with many conditional paths. I did go through this and it seems ok, the links are powered off correctly in this case as well (D0i3x for S0ix). But better is this intended behaviour is mentioned in the commit message, so people know what to expect.